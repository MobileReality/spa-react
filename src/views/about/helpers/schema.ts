import z from 'zod';

export const schema = z.object({
    firstName: z.string(),
});

export type FormSchema = z.infer<typeof schema>;
