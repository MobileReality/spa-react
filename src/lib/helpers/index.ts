export { isEven } from 'lib/helpers/is-even/is-even';
export { replaceReactAriaIds } from 'lib/helpers/replace-aria-react-ids';
