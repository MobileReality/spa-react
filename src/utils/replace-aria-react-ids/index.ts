export { replaceReactAriaIds } from './replace-aria-react-ids';
