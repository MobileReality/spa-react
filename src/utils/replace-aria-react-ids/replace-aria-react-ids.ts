const ariaSelector = (el: string) => `[${el}^="react-aria"]`;

export const replaceReactAriaIds = (container: HTMLElement) => {
    const selectors = ['id', 'for', 'aria-labelledby'];
    const regexp = /react-aria\d+-\d+/g;
    const staticId = 'static-id';

    /**
     * keep a map of the replaceIds to keep the matching between input "id" and label "for" attributes
     */
    const attributesMap: Record<string, string> = {};

    for (const [index, el] of container
        .querySelectorAll(selectors.map(ariaSelector).join(', '))
        .entries()) {
        for (const selector of selectors) {
            const attr = el.getAttribute(selector);

            if (attr?.match(regexp)) {
                const newAttr = `${staticId}-${index}`;

                el.setAttribute(selector, attributesMap[attr] || newAttr);

                attributesMap[attr] = newAttr;
            }
        }
    }
};
