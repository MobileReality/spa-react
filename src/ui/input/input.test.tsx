import React from 'react';
import { render } from '@testing-library/react';
import { replaceReactAriaIds } from 'helpers/replace-aria-react-ids';

import { Input } from './';

describe('Input', () => {
    it('should generate Input component snapshot', () => {
        const { container } = render(<Input label="Test Label" />);
        replaceReactAriaIds(container);
        expect(container).toMatchSnapshot();
    });
});
