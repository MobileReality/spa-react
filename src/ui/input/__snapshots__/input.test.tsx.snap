// Vitest Snapshot v1

exports[`Input > should generate Input component snapshot 1`] = `
<div>
  <div>
    <label
      for="static-id-0"
      id="static-id-0"
    >
      Test Label
    </label>
    <input
      aria-labelledby="static-id-0"
      defaultvalue=""
      id="static-id-0"
      type="text"
    />
  </div>
</div>
`;
